### PR TITLE
Pip-tools support

### PR DESCRIPTION
**Pull request by andyneff**
Add a recipe for tools to help make using pip-tools smooth

1. `pip-8210` - This is a "drop in replacement" for `pip` for when you want to pre-install a dependency using the existing `requirements.txt` file as a constraint.
2. `pip-fake` - This is the update for what was previously known as `fake_package`. If the capabilities need to grow, they will go in this recipe file.

### CODE REVIEW

File: pip-8210 [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+if [[ ${-} != *i* ]]; then
+  source_once &> /dev/null && return 0
+fi
+
+#*# pip-8210
+
+#**
+# .. file:: pip-8210
+#
+# This script allows us to install a package outside of pip-tools. In order to accomplish correctly, we use the ``requirements.txt`` file as a constraint files,and this and handle corner cases like pip issue 8210.
```
Comment (scott-vsi) [inline pip-8210:12]:
```suggestion
# This script allows us to install a package outside of pip-tools. In order to accomplish this correctly, we use the ``requirements.txt`` file as a constraint file to handle corner cases like pip issue 8210.
```

File: pip-8210 [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+#
+# It add the -c constraints to ``pip install``, ``pip download``, and ``pip wheel``
```
Comment (scott-vsi) [inline pip-8210:14]:
```suggestion
# Adds the -c constraints to ``pip install``, ``pip download``, and ``pip wheel``
```

Comment (scott-vsi) [reply pip-8210:14]: This doesn't really flow with anything else? What is it saying?

File: pip-8210 [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+#
+# You should activate your virtualenv before calling this; it will use the ``pip`` on the path.
+#
+# :Arguments: * [--if-not-found <executable name>] - Only runs the ``pip`` command if a specified executable is not found.
```
Comment (scott-vsi) [inline pip-8210:18]: Should this say "runs the specified executable if the pip command is not found?

File: pip-8210 [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  # function parse-generic(){ :;}
+  function parse-generic()
+  {
+    command_args=()
+    subcommand_args=()
+
+    while (( ${#} )); do
+      case "${1}" in
+        -*)
+          # Anything that takes one arguments from the array arguments_with_value
```
Comment (scott-vsi) [inline pip-8210:50]:
```suggestion
          # Anything that takes one argument from the array arguments_with_value
```

Comment (scott-vsi) [reply pip-8210:50]: Shouldn't arguments_with_value be documented as an input of some sort

File: pip-8210 (hunk 2 above)
Comment (andyneff) [reply pip-8210:14]:
```suggestion
# To accomplish this, it adds the ``-c`` constraints to ``pip install``, ``pip download``, and ``pip wheel`` sub-commands.
```

File: pip-8210 (hunk 3 above)
Comment (andyneff) [reply pip-8210:18]: No. It allows you to install `pip-tools` if `pip-sync` in not found. It's part of the chicken/egg resolution.

File: pip-8210 (hunk 4 above)
Comment (andyneff) [reply pip-8210:50]: This is just an exact copy from `command_tools.bsh` (for portability).

File: pip-8210 (hunk 3 above)
Comment (scott-vsi) [reply pip-8210:18]: Is this optional?

Comment (andyneff) [reply pip-8210:18]: Yes, there are `[]` for optional

**Pull request merged**